src/ohri-form.component.test.tsx

```diff
@@ -130,7 +132,36 @@ describe('OHRI Forms:', () => {
   });
 
   describe('Form submission', () => {
-    // TODO: Fillup test suite
+    it('should save data to the backend', async () => {
+      const abortControllerMock = new AbortController();
+
+      const mockSaveEncounter = jest.fn(saveEncounter);
+
+      const result = await mockSaveEncounter(abortControllerMock, mockVisit);
```

Comment:
Well, this isn't correct. You are passing a visit object to an API function that saves an encounter. That doesn't make sense. You needed to pass down an encounter. 

Ideally, this test was supposed to be an integral test. We should render a form, then using the screen and jest's support on user events, fill up the form and trigger a submission. We then assert that the`mockSaveEncounter` received an encounter that we expected. 

Just to test the transient behaviour, include an transient field in the form and assert that it's observation or isn't part of the encounter after submission.

Reply:
This is well understood.